spec-tests/src/main/java/org/candlepin/spec/bootstrap/client/PropertiesConfiguration.java

```diff
@@ -0,0 +1,181 @@
+/*
+ *  Copyright (c) 2009 - ${YEAR} Red Hat, Inc.
+ *
+ *  This software is licensed to you under the GNU General Public License,
+ *  version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ *  implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ *  FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ *  along with this software; if not, see
+ *  http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ *  Red Hat trademarks are not licensed under GPLv2. No permission is
+ *  granted to use or replicate Red Hat trademarks that are incorporated
+ *  in this software or its documentation.
+ */
+
+package org.candlepin.spec.bootstrap.client;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Provider;
+import javax.naming.ConfigurationException;
+import java.io.*;
+import java.nio.charset.Charset;
+import java.util.Enumeration;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PropertiesConfiguration{
+    private static final Logger log = LoggerFactory.getLogger(PropertiesConfiguration.class);
+    private Map<String, String> configMap = new ConcurrentHashMap<>();
```

Comment:
Do we need synchronized config? We will be loading it at the beginning and it should stay immutable for the whole run.

Reply:
See the above comment